### PR TITLE
Switch to JDK17 as a base runtime for action

### PR DESCRIPTION
**Pull request by denismakogon**

### CODE REVIEW

**Comment by denismakogon** [comment]
@sormuras, ready for review.

File: src/Download.java [hunk 1]
```diff
@@ -78,11 +78,12 @@ static void main(boolean dryRun, String... args) {
       // Emit warning when using an archived JDK build
       if (website.isArchivedUri(uri)) {
         GitHub.warn(
-            "JDK resolved to an archived build!\n"
-                + "These older versions of the JDK are provided\n"
-                + " to help developers debug issues in older systems.\n"
-                + "They are not updated with the latest security patches\n"
-                + " and are not recommended for use in production.");
+                """
```
Comment (sormuras) [inline src/Download.java:81]: 3 lines, one per sentence, would be nicer.

```java
"""
JDK resolved to an archived build!
These older versions of the JDK are provided to help developers debug issues in older systems.
They are not updated with the latest security patches and are not recommended for use in production.
"""
```

File: src/Download.java [hunk 2]
```diff
@@ -107,47 +108,34 @@ static void main(boolean dryRun, String... args) {
...
-
-    JDK(String release, String version, String os, String arch, String type) {
-      this.release = release;
-      this.version = version;
-      this.os = os;
-      this.arch = arch;
-      this.type = type;
-    }
+  record JDK(String release, String version, String os, String arch,
+             String type) {
```
Comment (sormuras) [inline src/Download.java:112]: Put all components in a single line.

```
record JDK(String release, String version, String os, String arch, String type) {
```

File: src/Download.java [hunk 3]
```diff
@@ -107,47 +108,34 @@ static void main(boolean dryRun, String... args) {
...
 
     @Override
-    public String toString() {
+      public String toString() {
```
Comment (sormuras) [inline src/Download.java:115]: Delete the custom `toString()` method. The one generated by `record` is fine.

**Review by sormuras — CHANGES_REQUESTED** [review]

File: src/Download.java (hunk 1 above)
Comment (denismakogon) [reply src/Download.java]: fixed

File: src/Download.java (hunk 3 above)
Comment (denismakogon) [reply src/Download.java]: fixed

**Review by sormuras — CHANGES_REQUESTED** [review]
Changes look good to me.

Now, all workflows in [.github/workflows](https://github.com/oracle-actions/setup-java/tree/main/.github/workflows) need a pass to use pre-installed Java 17 as well.

**Review by sormuras — APPROVED** [review]
Updated and run the CI workflow test. We're ready to switch, methinks.

**Pull request merged**